General queries/Baseline Comparison.txt

```diff
@@ -0,0 +1,239 @@
+    | summarize EventTime=max(EventTime), Count=count(), AdditionalData=makeset(FolderPath) by FileName, MachineId
+    // Replace various mbam executables that are semiunique-generated with some text to help reduce noise
+    | project EventTime, Entity=FileName, Count, AdditionalData=tostring(AdditionalData), MachineId, DataType="Process Creation"
+};
+// Function to get powershell commands given a list of computer names and how far back to look
+let GetPSCommands=(InMachineId: dynamic, LeftEventTime: datetime) {
+    MiscEvents
+    | where "PowerShell Command" in (ReturnSets)
+    | where EventTime > LeftEventTime
+    | where MachineId in~ (InMachineId)
+    | where ActionType == 'PowerShellCommand'
+    // Remove two different signatures for scripts being executed which cause a lot of noise
+    // The first signature matches scripts generated as part of testing execution policy
+    // The second signature matches scripts generated by SCCM
+    | where not(AdditionalFields matches regex @"Script_[0-9a-f]{20}")
+    | where not(AdditionalFields matches regex @"[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}")
+    | summarize EventTime=max(EventTime), count(), IPFN_Set=makeset(InitiatingProcessFileName) by AdditionalFields, MachineId
+    | project EventTime, Entity=tostring(extractjson("$.Command", AdditionalFields)), Count=count_, AdditionalData=tostring(IPFN_Set), MachineId, DataType="PowerShell Command"
+};
+// Function to get file creations given a list of computer names and how far back to look
+let GetFileCreates=(InMachineId: dynamic, LeftEventTime: datetime) {
+    FileCreationEvents
+    | where "File Creation" in (ReturnSets)
+    | where EventTime > LeftEventTime
+    | where MachineId in~ (InMachineId)
+    // Remove temporary files created by office products
+    | where not(FileName matches regex @"~.*\.(doc[xm]?|ppt[xm]?|xls[xm]?|dotm|rtf|xlam|lnk)")
+    // Replace two different signatures for PS scripts being created which cause a lot of noise
+    | extend iff(FileName matches regex @"[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\.ps1" or
+                 FileName matches regex @"[0-9a-z]{8}\.[0-9a-z]{3}\.ps1", "RANDOM.ps1", FileName)
+    | summarize EventTime=max(EventTime), FP_Set=makeset(FolderPath), count() by FileName, MachineId
+    | project EventTime, Entity=FileName, Count=count_, AdditionalData=tostring(FP_Set), MachineId, DataType="File Creation"
+};
+// Function to get logon events given a list of computer names and how far back to look
+let GetLogonEvents=(InMachineId: dynamic, LeftEventTime: datetime) {
+    LogonEvents
+    | where "Logon" in (ReturnSets)
+    | where EventTime > LeftEventTime
+    | where MachineId in~ (InMachineId)
+    // Remove logons made by WDM or UMFD
+    | where AccountDomain !in ('font driver host', 'window manager')
+    // Replace qualys logons which cause a lot of noise
```

Comment:
Seems DSRE specific to me

Reply:
Is DSRE-specific, removed in latest commit.